### PR TITLE
Fix #1428 - zstdgrep returned 1 on both matched and unmatched file

### LINKED ISSUE #1428: zstdgrep returns 0 on non-match
The return code of grep is often used—particularly with "grep -q", but not always—in a script to detect the presence or absence of a match.  And zgrep preserves this behavior.  But zstdgrep does not:
```
$ echo a | gzip | zgrep a; echo $?
a
0
$ echo a | gzip | zgrep b; echo $?
1
$ echo a | zstd | zstdgrep a; echo $?
a
0
$ echo a | zstd | zstdgrep b; echo $?
0
```
Even with "zstdgrep -q", a non-match still returns 0.
I'd recommend bringing zstdgrep in line.

### PR DESCRIPTION
**Pull request by lzutao**

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -355,6 +355,10 @@ test-gzstd: gzstd
 test-zstdgrep: gzstd
 	@echo a | $(PRGDIR)/zstd | $(PRGDIR)/zstdgrep a
 	@echo a | $(PRGDIR)/zstd | $(PRGDIR)/zstdgrep b && return 1 || return 0
+	-[ -f zstdcat ] || ln -s $(PRGDIR)/zstd zstdcat
```
Comment (Cyan4973) [inline tests/Makefile]: The test seems to leave a `zstdcat` link in the working directory.
Is that intentional and beneficial ?
The links seems only useful for the test itself.

Comment (lzutao) [reply tests/Makefile]: The `test-lz4` test does leave `lz4` in cwd too. But I'd move `zstdcat` link to `/tmp`.

**Comment by lzutao** [comment]
@Cyan4973 Off-topic, I think we could run `scan-build` with meson and ninja. 

**Pros**:
- It is faster than `make staticAnalyze`
- It is directly supported by meson and ninja

**Cons**:
- Some test files (`.c`) are not included in Meson build

File: tests/Makefile (hunk 1 above)
Comment (Cyan4973) [reply tests/Makefile]: Yep, that's a good choice

**Comment by Cyan4973** [comment]
I agree on the `meson` `scan-build` proposal.

The current `make staticAnalyze` is way too long, 
because it analyzes the same files multiple times (the ones which belong to multiple projects). 
It would need to be refactored.

**Comment by lzutao** [comment]
Oops, there's an error in `test-zstdgrep` when creating link in `/tmp`.
Let's me fix this.

**Pull request merged**